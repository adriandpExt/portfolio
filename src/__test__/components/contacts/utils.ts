import { fireEvent } from "@testing-library/react";

export const changeInputValue = (
  container: HTMLElement,
  testId: string,
  value: string
) => {
  const inputElement = container.querySelector(
    `[data-testid="${testId}"] input`
  );
  if (!inputElement) {
    return new Error(`Input element with data-testid "${testId}" not found`);
  }
  fireEvent.change(inputElement, { target: { value } });
};

export const changeTextareaValue = (
  container: HTMLElement,
  testId: string,
  value: string
) => {
  const inputElement = container.querySelector(
    `[data-testid="${testId}"] textarea`
  );
  if (!inputElement) {
    return new Error(`Input element with data-testid "${testId}" not found`);
  }
  fireEvent.change(inputElement, { target: { value } });
};
