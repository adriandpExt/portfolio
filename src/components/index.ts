export * from "./appbar";
export * from "./svg-icons";
export * from "./aside";
