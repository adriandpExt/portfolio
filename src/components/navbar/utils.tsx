import AssignmentIndIcon from "@mui/icons-material/AssignmentInd";
import HomeIcon from "@mui/icons-material/Home";
import AppsIcon from "@mui/icons-material/Apps";
import ContactMailIcon from "@mui/icons-material/ContactMail";
import InfoIcon from "@mui/icons-material/Info";

export const menuItems = [
  { listIcon: <HomeIcon />, listText: "Home", listPath: "/portfolio " },
  {
    listIcon: <AssignmentIndIcon />,
    listText: "Resume",
    listPath: "/portfolio/cv",
  },
  {
    listIcon: <InfoIcon />,
    listText: "About",
    listPath: "/portfolio/about",
  },
  {
    listIcon: <AppsIcon />,
    listText: "Projects",
    listPath: "/portfolio/project",
  },
  {
    listIcon: <ContactMailIcon />,
    listText: "Contact",
    listPath: "/portfolio/contact",
  },
];
