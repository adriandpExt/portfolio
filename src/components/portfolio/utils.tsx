// import avatar from "../../assets/image/avatar.png";
import empf1 from "../../assets/empf/empf1.png";
import empf2 from "../../assets/empf/empf2.png";
import empf3 from "../../assets/empf/empf3.png";

export const projects = [
  {
    name: "EMPF",
    description: `EMPF is a centralised electronic platform that aims to standardise and automate the administration processes of MPF schemes. An electronic platform to be launched by the Mandatory Provident Fund Schemes Authority (MPFA), it will cover all 12 MPF trustees, as well as the 357,000 employers and 4.7 million members that Hong Kong's mandatory pension scheme covers. Collaboration with other developer`,
    image: [empf1, empf2, empf3],
  },
];
