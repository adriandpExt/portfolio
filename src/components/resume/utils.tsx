export const details = [
  {
    jobtitle: "Solutions Developer",
    companyName: "LENOVO PCCW SOLUTIONS PHILIPPINES INC",
    year: "2022-2024",
    description: `Motivated Solution Developer with a dedication to delivering sophisticated and effective web applications. I am in search of a challenging position where I can utilize my two years of expertise in front-end development and UI design, employing contemporary technologies to contribute to the company's success`,
  },
];
