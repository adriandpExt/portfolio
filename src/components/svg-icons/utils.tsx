export type IconName =
  | "ic_gmail"
  | "ic_mobile"
  | "ic_facebook"
  | "ic_github"
  | "ic_linkedin"
  | "ic_vue"
  | "ic_react"
  | "ic_html"
  | "ic_css"
  | "ic_javascript"
  | "ic_bootstrap"
  | "ic_material_ui"
  | "ic_tailwind";
