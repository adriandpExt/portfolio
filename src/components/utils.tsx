export const toggleDarkTheme = (isDarkMode: boolean) => {
  return isDarkMode ? "bg-gray-900 text-white" : "bg-gray-200 text-gray-800";
};

export const toggleLightTheme = (isDarkMode: boolean) => {
  return isDarkMode ? "bg-gray-200 text-gray-800" : "bg-gray-900 text-white";
};

export const handleToggleDropIcon = (isDarkMode: boolean) => {
  return isDarkMode ? "bg-gray-200  text-gray-800" : "bg-gray-900  text-white";
};
