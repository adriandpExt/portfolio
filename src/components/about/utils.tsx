import { IconName } from "../svg-icons/utils";

export const techStack: IconName[] = [
  "ic_react",
  "ic_vue",
  "ic_javascript",
  "ic_typescript",
  "ic_html",
  "ic_css",
  "ic_bootstrap",
  "ic_tailwind",
  "ic_material_ui",
];
export const techTools: IconName[] = ["ic_git", "ic_mysql", "ic_vscode"];
