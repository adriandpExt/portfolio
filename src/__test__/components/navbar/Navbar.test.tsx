import "@testing-library/jest-dom";
import { Navbar } from "../../../components";
import { render, waitFor } from "@testing-library/react";
import { menuItems } from "../../../components/navbar/utils";

jest.mock("react-router-dom");

describe("navbar component test", () => {
  test("Navbar renders menu items correctly", () => {
    const { container } = render(<Navbar />);

    expect(container).toMatchSnapshot();
  });

  test("should render info", () => {
    const { getByTestId } = render(<Navbar />);
    waitFor(() => expect(getByTestId("email")).toBeInTheDocument());
    waitFor(() => expect(getByTestId("phoneNumber")).toBeInTheDocument());
  });

  test("should render list", () => {
    menuItems.map((item) => {
      waitFor(() => expect(item.listText).toBeInTheDocument());
    });
  });
});
