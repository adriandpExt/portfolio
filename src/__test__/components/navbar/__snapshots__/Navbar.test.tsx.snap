// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navbar component test Navbar renders menu items correctly 1`] = `
<div>
  <nav
    class="MuiBox-root css-0"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-1d1jaor-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tp2qox-MuiSvgIcon-root"
            data-testid="AppsIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M4 8h4V4H4zm6 12h4v-4h-4zm-6 0h4v-4H4zm0-6h4v-4H4zm6 0h4v-4h-4zm6-10v4h4V4zm-6 4h4V4h-4zm6 6h4v-4h-4zm0 6h4v-4h-4z"
            />
          </svg>
          <h5
            class="MuiTypography-root MuiTypography-h5 css-5eczce-MuiTypography-root"
            data-testid="nameLogo"
          >
            ADRIAN
          </h5>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
  </nav>
</div>
`;
